source/io.c

```diff
@@ -259,6 +202,16 @@ PyObject *aws_py_io_client_tls_ctx_new(PyObject *self, PyObject *args) {
     if (certificate_buffer && private_key_buffer && certificate_buffer_len > 0 && private_key_buffer_len > 0) {
         struct aws_byte_cursor cert = aws_byte_cursor_from_array(certificate_buffer, certificate_buffer_len);
         struct aws_byte_cursor key = aws_byte_cursor_from_array(private_key_buffer, private_key_buffer_len);
+        
+       /* adding the null terminator in python land screws up the byte buf apis. We never consider the null terminator
```

Comment:
Ah, fuck, that's my fault. How did this ever work ever?

Reply:
it worked fine till we started loading certs on apple, and it did not like that NULL at all